ignite/pkg/cosmosclient/cosmosclient_test.go

```diff
@@ -0,0 +1,369 @@
+package cosmosclient
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ignite-hq/cli/ignite/pkg/cosmosclient/testutil"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/require"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	tmtypes "github.com/tendermint/tendermint/types"
+)
+
+func TestGetBlockTXs(t *testing.T) {
+	m := testutil.NewTendermintClientMock(t)
+	ctx := context.Background()
+
+	// Mock the Block RPC endpoint
+	block := createTestBlock(1)
+
+	m.On("Block", ctx, &block.Height).Return(&ctypes.ResultBlock{Block: &block}, nil)
+
+	// Mock the TxSearch RPC endpoint
+	searchQry := createTxSearchByHeightQuery(block.Height)
```

Comment:
I beleive we shouldn't depend on private decl on unit testing, this has quite a lot pros including; we don't need to change tests when implementation details change, we're forced to design a more pluggable API/better abstraction

Reply:
In this specific case, I would even build the Q manually with `fmt.Sprintf`, tests doesn't need to be fully programmatic. This way we can also test that if our logic in `createTxSearchByHeightQuery` is correct.